### PR TITLE
Add Bakeneko 80 keyboard

### PR DESCRIPTION
**Pull request by kkatano**
## Description

Add new keyboard named Bakeneko 80.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bakeneko80/config.h [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x2FCC
+#define PRODUCT_ID 0x8DEF
```
Comment (fauxpark) [inline keyboards/bakeneko80/config.h:24]: Where do these values come from?

File: keyboards/bakeneko80/config.h [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#ifndef LINK_TIME_OPTIMIZATION_ENABLE
+  #define NO_ACTION_MACRO
+  #define NO_ACTION_FUNCTION
+#endif
```
Comment (fauxpark) [inline keyboards/bakeneko80/config.h:198]:
```suggestion
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
```

File: keyboards/bakeneko80/rules.mk [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/bakeneko80/rules.mk:23]:
```suggestion
```

File: keyboards/bakeneko80/config.h (hunk 1 above)
Comment (kkatano) [reply keyboards/bakeneko80/config.h:24]: It was generated by following command.

```
$ echo -n "Koichi Katano" | shasum | cut -c1-4
2fcc

$ echo -n "Bakeneko 80" | shasum | cut -c1-4
8def
```

Comment (fauxpark) [reply keyboards/bakeneko80/config.h:24]: In that case...
```suggestion
#define VENDOR_ID 0xFEED
#define PRODUCT_ID 0x8DEF
```

Comment (kkatano) [reply keyboards/bakeneko80/config.h:24]: Oh, I see.

**Pull request merged**